File: product-of-array-except-self/product-of-array-except-self.go
func productExceptSelf(nums []int) []int {
    n := len(nums)
    
    arr1 := make([]int, n)
    arr2 := make([]int, n)
    arr3 := make([]int, n)
    
    arr1[0] = 1
    arr2[n-1] = 1
    
    for i:=1; i<n; i++{
        arr1[i] = arr1[i-1] * nums[i-1]
    }
    for i:= n-2; i>=0; i--{
        arr2[i] = arr2[i+1] * nums[i+1]
    }
    for i:=0; i<n; i++{
        arr3[i] = arr1[i] * arr2[i]
    }
    return arr3
}

File: product-of-array-except-self/product-of-array-except-self.py
class Solution(object):
    def productExceptSelf(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        n = len(nums)
        nums1 = [1] * n
        nums2 = [1] * n
        
        for i in range(1, n):
            nums1[i] = nums[i-1] * nums1[i-1]
        for i in range(n-2, -1, -1):
            nums2[i] = nums[i+1] * nums2[i+1]
        for i in range(n):
            nums2[i] = nums1[i] * nums2[i]
            
        return nums2